frontend/packages/yki/src/components/registration/examSession/PublicExamSessionListingRow.tsx

```diff
@@ -215,54 +212,21 @@ export const PublicExamSessionListingRow = ({
   const { t } = usePublicTranslation({
     keyPrefix: 'yki.component.registration.registrationButtonLabels',
   });
-  const now = dayjs();
   const { isPhone } = useWindowProperties();
 
   const locationInfo = ExamSessionUtils.getLocationInfo(
     examSession,
     getCurrentLang()
   );
-  const registrationPeriodOpen = ExamSessionUtils.isRegistrationOpen(
-    examSession,
-    now
-  );
-  const postAdmissionOpen = ExamSessionUtils.isPostAdmissionOpen(
-    examSession,
-    now
-  );
-  const relevantPeriod =
-    ExamSessionUtils.getCurrentOrFutureAdmissionPeriod(examSession);
-  const getAvailablePlacesText = () => {
-    if (
-      relevantPeriod.kind === RegistrationKind.Admission &&
-      examSession.participants < examSession.max_participants
-    ) {
-      return `${
-        examSession.max_participants - (examSession.participants ?? 0)
-      }`;
-    } else if (
-      relevantPeriod.kind === RegistrationKind.PostAdmission &&
-      examSession.post_admission_quota &&
-      examSession.pa_participants < examSession.post_admission_quota
-    ) {
-      return `${
-        examSession.post_admission_quota - (examSession.pa_participants ?? 0)
-      }`;
-    }
 
-    return '' + t('full');
-  };
+  const { open, availablePlaces, availableQueue, participants, quota } =
+    ExamSessionUtils.getCurrentOrFutureAdmissionPeriod(examSession);
+  const remainingPlaces = Math.max(quota - participants, 0);
+  const availablePlacesText =
+    remainingPlaces > 0 ? '' + remainingPlaces : t('full');
 
-  const availablePlacesText = getAvailablePlacesText();
   const registerActionAvailable =
-    examSession.open &&
-    ((registrationPeriodOpen &&
-      examSession.participants < examSession.max_participants) ||
-      (registrationPeriodOpen && !examSession.queue_full) ||
-      (postAdmissionOpen &&
-        examSession.post_admission_quota !== undefined &&
-        examSession.post_admission_quota > 0 &&
-        examSession.pa_participants < examSession.post_admission_quota));
+    examSession.open && open && (availablePlaces || availableQueue);
```

Comment:
Mikä tän `open` kentän käyttö muuten mahtoikaan olla tai oliko sillä jotain oikeaa funktiota? Backendin palauttama tieto siitä onko sessio sen mielestä avoinna rekisteröinneille? Mietin jos frontin päässä hanskataan tuo tsekkaus onko tilaisuus avoinna vai ei niin olisko parempi ettei tälle tehtäisi mitään tai olisi välttis ees interfacessa määritelty.

Reply:
Joo, tosiaan backendin palauttama tieto. Mutta tosiaan pitäisi ehkä ottaa pois tältä riviltä ja tehdä selkeä päätös siitä että luotetaanko nimenomaan backendin palauttamaan dataan vai frontendin laskelmiin. Erona tietysti että frontendissä laskettuna voi aukiolo vaihtua rendauksien välillä, backendiin luotettaessa vain kun haetaan uudet datat..

Reply:
Lähtökohtaisesti siis mielestäni voisi luottaa siihen mitä frontendissa on laskettu, mutta pitäisi ehkä vielä ottaa jotenkin huomioon käyttäjän selaimen aikavyöhyke näitä laskettaessa.